src/lightning/pytorch/strategies/hpu_parallel.py

```diff
@@ -123,16 +122,6 @@ def optimizer_step(
         htcore.mark_step()
         return optimizer_output
 
-    def validation_step_end(self, step_output: STEP_OUTPUT) -> STEP_OUTPUT:
-        # Break lazy accumulation of graph after every step
-        htcore.mark_step()
-        return step_output
-
-    def test_step_end(self, step_output: STEP_OUTPUT) -> STEP_OUTPUT:
-        # Break lazy accumulation of graph after every step
-        htcore.mark_step()
```

Comment:
Isn't this still necessary?

Reply:
No, since we don't go through the step_end code path anymore, so there aren't any graph operations there to mark a step for. 

mark_step is called also in on_after_backward and optimizer_step() hook as required by habana. See: https://docs.habana.ai/en/latest/PyTorch/PyTorch_Model_Porting/Porting_PyTorch_Models_to_Gaudi.html?highlight=mark_step#porting-pytorch-models-to-gaudi